src/api.jl

```diff
@@ -1,55 +1,298 @@
+
+Return `true` if the periodic orbits `po1` and `po2` are equal within the given thresholds.
+
+## Keyword arguments
+
+* `Tthres` : distance between periodic orbits must be less than this threshold
+* `dthres` : difference in periods of the periodic orbits must be less than this threshold
+* `distance` : distance function used to compute the distance between the periodic orbits
+
+Distance between the orbits is computed using the given distance function `distance`.
+The default distance function is `StrictlyMinimumDistance(true, Euclidean())` which finds the minimal 
+Euclidean distance between any pair of points where one point belongs to `po1` and the other to `po2``. 
+For other options of the distance function, see `StateSpaceSets.set_distance`.
+Custom distance function can be provided as well.
+"""
+function poequal(
+        po1::PeriodicOrbit, po2::PeriodicOrbit;
+        Tthres = 1e-3,
+        dthres = 1e-3,
+        distance = StrictlyMinimumDistance(true, Euclidean())
+    )
+    if abs(po1.T - po2.T) > Tthres
+        return false
+    end
+    d = podistance(po1, po2, distance)
+    return d < dthres
+end
+
+"""
+    minimal_period(ds::DynamicalSystem, po::PeriodicOrbit, atol=1e-4) → minT_po
+
+Compute the minimal period of the periodic orbit `po` of the dynamical system `ds`.
+Return the periodic orbit `minT_po` with the minimal period. 
+
+For discrete systems, a valid period would be any natural multiple of the minimal period. 
+Hence, all natural divisors of the period `po.T` are checked as a potential period. 
+A point `u0` of the periodic orbit `po` is iterated `n` times and if the distance between the initial point `u0` 
+and the final point is less than `atol`, the period of the orbit is `n`.
+
+For continuous systems, the minimal period check is not implemented yet. 
+The function returns a periodic orbit `minT_po` which a copy of input periodic orbit `po`.
+"""
+function minimal_period(ds::DynamicalSystem, po::PeriodicOrbit, atol=1e-4)
+    type1 = isdiscretetime(ds)
+    type2 = isdiscretetime(po)
+    if type1 == type2
+        return _minimal_period(ds, po, atol)
+    else
+        throw(ArgumentError("Both the periodic orbit and the dynamical system have to be either discrete or continuous."))
+    end
+end
 
-@kwdef struct Algorithm1 <: PeriodicOrbitFinder
-    param1 = 1
-    param2 = 2
-    param3 = 3
+function _minimal_period(ds::DiscreteTimeDynamicalSystem, po::PeriodicOrbit, atol)
+    u = po.points[1]
+    for n in 1:po.T-1
+        po.T % n != 0 && continue
+        reinit!(ds, u)
+        step!(ds, n)
+        if norm(u - current_state(ds)) < atol
+            points = complete_orbit(ds, u, n; Δt=1)
+            minT_po = PeriodicOrbit(points, n, po.stable)
+            return minT_po
+        end
+    end
+    return po
 end
 
+function _minimal_period(ds::ContinuousTimeDynamicalSystem, po::PeriodicOrbit, atol)
+    # TODO
+    minT_po = po
+    return minT_po
+end
```

Comment:
I have a poincare map version of this function that I can contribute some time later.